src/bin/cli/mod.rs

```diff
@@ -515,27 +515,20 @@ impl PackageTrait for DocPackage {
 pub fn imports_arg<T: PackageTrait>(package: &T) -> FileResolver {
     let mut resolver = FileResolver::default();
 
-    for filename in package.get_input() {
-        if let Ok(path) = PathBuf::from(filename).canonicalize() {
-            let _ = resolver.add_import_path(path.parent().unwrap());
-        }
-    }
-
+    let mut added_path = false;
     if let Some(paths) = package.get_import_path() {
         for path in paths {
-            if let Err(e) = resolver.add_import_path(path) {
-                eprintln!("error: import path '{}': {}", path.to_string_lossy(), e);
-                exit(1);
-            }
+            resolver.add_import_path(path);
+            added_path = true;
         }
     }
+    if !added_path {
+        resolver.add_import_path(&PathBuf::from(""));
+    }
 
     if let Some(maps) = package.get_import_map() {
         for (map, path) in maps {
-            if let Err(e) = resolver.add_import_map(OsString::from(map), path.clone()) {
-                eprintln!("error: import path '{}': {}", path.display(), e);
-                exit(1);
-            }
+            resolver.add_import_map(OsString::from(map), path.clone());
```

Comment:
We should warn if we're overwriting a previous set mapping.

Reply:
As you [mentioned before](https://github.com/hyperledger/solang/pull/1487#discussion_r1304103625), we will need to warn from solang.rs, not from here. Not sure how to bubble this data up to solang.rs, maybe do a separate check there?

FWIW Solc doesn't issue a warning, so if it's a headache to implement maybe better to not?

Reply:
Okay, I tried to add an `eprintln` and it's not showing up on stderr. Your call on how you want to implement this.